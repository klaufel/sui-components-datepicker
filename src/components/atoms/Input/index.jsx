import React from 'react'
import PropTypes from 'prop-types'
import './index.scss'

const Input = ({...props}) => {
  return (
    <div className="FormField">
      <input className="FormField-input" {...props} />
    </div>
  )
}

export default Input
