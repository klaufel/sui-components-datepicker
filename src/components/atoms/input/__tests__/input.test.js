import React from 'react'
import {render} from '@testing-library/react'
import Input from '..'

describe('Input', () => {
  test('it works default', () => {
    const {container} = render(<Input />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with id and name attr', () => {
    const {container} = render(<Input id="id" name="name" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with placeholder', () => {
    const {container} = render(<Input placeholder="Placeholder" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with default value', () => {
    const {container} = render(
      <Input placeholder="Placeholder" value="Default value" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with disabled attr', () => {
    const {container} = render(
      <Input placeholder="Placeholder" value="Default value" disabled />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with readonly attr', () => {
    const {container} = render(
      <Input placeholder="Placeholder" value="Default value" readonly />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
