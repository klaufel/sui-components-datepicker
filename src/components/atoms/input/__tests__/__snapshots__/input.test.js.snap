// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input it works default 1`] = `
<input
  class="Input"
  value=""
/>
`;

exports[`Input it works with default value 1`] = `
<input
  class="Input"
  placeholder="Placeholder"
  value="Default value"
/>
`;

exports[`Input it works with disabled attr 1`] = `
<input
  class="Input"
  disabled=""
  placeholder="Placeholder"
  value="Default value"
/>
`;

exports[`Input it works with id and name attr 1`] = `
<input
  class="Input"
  id="id"
  name="name"
  value=""
/>
`;

exports[`Input it works with placeholder 1`] = `
<input
  class="Input"
  placeholder="Placeholder"
  value=""
/>
`;

exports[`Input it works with readonly attr 1`] = `
<input
  class="Input"
  placeholder="Placeholder"
  readonly=""
  value="Default value"
/>
`;
